[minio] Update to 2018-09-01T00-38-25Z, add tests

predominant opened the pull request:
Signed-off-by: Graham Weldon <graham@grahamweldon.com>

Testing:

```
hab studio enter
./minio/tests/test.sh
```

Sample output:

```
 ✓ Command is on path
 ✓ Version matches, via help output
 ✓ Service is running
 ✓ A single process
 ✓ Listening on port 9000

5 tests, 0 failures
```

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

smacfarlane commented:
Thanks for this PR! We're currently in the process of a base plans refresh #1767 and hope to be done later this week. I'd like to hold off on merging this until that is done. Are there any changes here you're waiting on to unblock your work?

predominant commented:
Happy to wait.

nellshamrell commented on minio/test.bats:
```diff
@@ -0,0 +1,25 @@
+source ./plan.sh
+
+@test "Command is on path" {
+  [ "$(command -v gnatsd)" ]
```
should this be minio, rather than gnatsd?

predominant replied:
Copy paste!

predominant commented:
@nellshamrell Fixed

eeyun commented:
Personally I think I'd prefer to see these test get put into a tests directory rather than clutter the root of the package path.

bdangit commented:
I think that would be a cleaner presentation.  It would also lend itself to easily build automated plan testing later on.

predominant commented:
Updated @eeyun and @bdangit 
Excellent suggestion, and its done now.

Also updated the PR description to include new test instructions.

![tenor-170516616](https://user-images.githubusercontent.com/24568/45248599-ea539500-b34d-11e8-9163-03a72c4510a7.gif)

predominant commented:
If this PR goes through, I'll make a sweep back through all other tests i've added, and standardise.

nellshamrell commented:
Built and promoted!
